src/Microsoft.ComponentDetection.Common/DependencyGraph/ComponentRecorder.cs

```diff
@@ -96,10 +114,17 @@ public sealed class SingleFileComponentRecorder : ISingleFileComponentRecorder
 
         private readonly ConcurrentDictionary<string, DetectedComponent> detectedComponentsInternal = new ConcurrentDictionary<string, DetectedComponent>();
 
+        /// <summary>
+        /// Dictionary of components which had an error during parsing and a dummy data value that only allocates 1 byte.
+        /// </summary>
+        private readonly ConcurrentDictionary<string, byte> skippedComponentsInternal = new ConcurrentDictionary<string, byte>();
```

Comment:
Why not just use a `ConcurrentBag<string>`?

Reply:
This is so we aren't iterating the entire list when searching for duplicates on list addition - this is fine for repos with a small number of components but can get expensive for large repositories.

Reply:
forgot .NET doesn't like concurrent hash sets :(